### PR TITLE
cleanup hardcode hacks

### PR DESCRIPTION
**Pull request by jsturdy**
## Description
* Remove several hard-coded hacks

### Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Motivation and Context
Makes code more flexible by default

## How Has This Been Tested?
Operating at P5

* When this is rebase+merged I'll put it on `master`, and create the `1.0.0` tag
 * Only a few P5 operation specific code will then go into this tag series
* Then will create the new branch for our `2.0.0` release line
 *  prerel tag will probably be `1.99.0`
 * Code for clean and proper `xhal` inclusion will be going into this branch

### CODE REVIEW

File: gemhardware/include/gem/hw/glib/GLIBManager.h [hunk 1]
```diff
@@ -87,6 +87,7 @@ namespace gem {
             xdata::Integer crateID;
             xdata::Integer slotID;
             xdata::String  cardName;
+            xdata::String  birdName;
```
Comment (bdorney) [inline gemhardware/include/gem/hw/glib/GLIBManager.h]: what is the difference between `birdName` and `cardName`?

Comment (bdorney) [reply gemhardware/include/gem/hw/glib/GLIBManager.h]: I see thanks

Comment (bdorney) [reply gemhardware/include/gem/hw/glib/GLIBManager.h]: Maybe we could add a commented line here to illustrate these distinctions?

File: gemsupervisor/src/common/GEMSupervisor.cc [hunk 2]
```diff
@@ -384,8 +384,6 @@ void gem::supervisor::GEMSupervisor::configureAction()
     m_globalState.update();
   }
 
-  pausePhaseMonitoring();
```
Comment (mexanick) [inline gemsupervisor/src/common/GEMSupervisor.cc]: What was the reason to pause phase monitoring an was this change tested?

Comment (jsturdy) [reply gemsupervisor/src/common/GEMSupervisor.cc]: The `pause`/`resume` was implemented when we didn't have the `memhub` protection against potentially colliding transactions.

**Comment by jsturdy** [comment]
Actually, based on the problem of today, I want to add three more things to this PR, though they can probably wait for the "maintenance" branch:
* SCA `HardReset` commands
* VFAT watchdog while running
* monitoring of links, and removing from the active links list, links which have lost communication
 * if the SCA `HardReset` works, this could be attempted, and a subsequent reconfiguration
 * if the SCA `HardReset` fails, this link would be excluded from any ongoing configuration (if the config was destroyed, this information would be reset)
* possibly also trigger `RunningDegraded` if the work is not too much for both the link lost issue, as well as any link `autokilled`

**Pull request merged**